app/code/Magento/InventoryCatalogAdminUi/Ui/DataProvider/Product/Listing/Modifier/QuantityPerSource.php

```diff
@@ -41,22 +44,38 @@ class QuantityPerSource extends AbstractModifier
      */
     private $getSourceItemsBySku;
 
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var SourceItemRepositoryInterface
+     */
+    private $sourceItemRepository;
+
     /**
      * @param IsSingleSourceModeInterface $isSingleSourceMode
      * @param IsSourceItemManagementAllowedForProductTypeInterface $isSourceItemManagementAllowedForProductType
      * @param SourceRepositoryInterface $sourceRepository
      * @param GetSourceItemsBySkuInterface $getSourceItemsBySku
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param SourceItemRepositoryInterface $sourceItemRepository
```

Comment:
all the new dependencies being added to the existing classes should be added as optional (nullable) to preserve Backward Compatibility 

This rule does not apply for Plugins and Observers

@IvanPletnyov please fix this in this and other PRs